platform/security/handler/security-handler-api/src/main/java/org/codice/ddf/security/handler/api/PKIAuthenticationTokenFactory.java

```diff
@@ -29,20 +36,25 @@
 
   private Merlin merlin;
 
-  private String signaturePropertiesPath;
-
   /** Initializes Merlin crypto object. */
   public void init() {
-    try {
-      merlin =
-          new Merlin(
-              PropertiesLoader.loadProperties(signaturePropertiesPath),
-              PKIAuthenticationTokenFactory.class.getClassLoader(),
-              null);
-    } catch (WSSecurityException | IOException e) {
+    // NOTE: THE TRUSTSTORE SHOULD BE USED FOR CERTIFICATE VALIDATION!!!!
+    Path trustStorePath = Paths.get(SecurityConstants.getTruststorePath());
+
+    if (!trustStorePath.isAbsolute()) {
+      Path ddfHomePath = Paths.get(System.getProperty("ddf.home"));
+      trustStorePath = Paths.get(ddfHomePath.toString(), trustStorePath.toString());
+    }
+
```

Comment:
✏️ Might want to add `doPrivileged` calls around the `InputStream` initialization and the `truststore.load` down below.